test/e2e/framework/psp_util.go

```diff
@@ -0,0 +1,141 @@
+			HostNetwork:              true,
+			HostPorts:                []extensionsv1beta1.HostPortRange{{Min: 0, Max: 65535}},
+			HostIPC:                  true,
+			HostPID:                  true,
+			RunAsUser: extensionsv1beta1.RunAsUserStrategyOptions{
+				Rule: extensionsv1beta1.RunAsUserStrategyRunAsAny,
+			},
+			SELinux: extensionsv1beta1.SELinuxStrategyOptions{
+				Rule: extensionsv1beta1.SELinuxStrategyRunAsAny,
+			},
+			SupplementalGroups: extensionsv1beta1.SupplementalGroupsStrategyOptions{
+				Rule: extensionsv1beta1.SupplementalGroupsStrategyRunAsAny,
+			},
+			FSGroup: extensionsv1beta1.FSGroupStrategyOptions{
+				Rule: extensionsv1beta1.FSGroupStrategyRunAsAny,
+			},
+			ReadOnlyRootFilesystem: false,
+		},
+	}
+}
+
+func IsPodSecurityPolicyEnabled(f *Framework) bool {
+	isPSPEnabledOnce.Do(func() {
+		psps, err := f.ClientSet.ExtensionsV1beta1().PodSecurityPolicies().List(metav1.ListOptions{})
+		if err != nil {
+			Logf("Error listing PodSecurityPolicies; assuming PodSecurityPolicy is disabled: %v", err)
+			isPSPEnabled = false
+		} else if psps == nil || len(psps.Items) == 0 {
+			Logf("No PodSecurityPolicies found; assuming PodSecurityPolicy is disabled.")
+			isPSPEnabled = false
+		} else {
+			Logf("Found PodSecurityPolicies; assuming PodSecurityPolicy is enabled.")
+			isPSPEnabled = true
+		}
+	})
+	return isPSPEnabled
+}
+
+var (
+	privilegedPSPOnce sync.Once
+)
+
+func CreatePrivilegedPSPBinding(f *Framework, namespace string) {
+	if !IsPodSecurityPolicyEnabled(f) {
+		return
+	}
+	// Create the privileged PSP & role
+	privilegedPSPOnce.Do(func() {
+		priv, _ := f.ClientSet.ExtensionsV1beta1().PodSecurityPolicies().Get(
+			podSecurityPolicyPrivileged, metav1.GetOptions{})
+		if priv != nil {
+			// Privileged PSP was already created.
+			return
+		}
+
+		psp := PrivilegedPSP(podSecurityPolicyPrivileged)
+		psp, err := f.ClientSet.ExtensionsV1beta1().PodSecurityPolicies().Create(psp)
+		ExpectNoError(err, "Failed to create PSP %s", podSecurityPolicyPrivileged)
+
+		// Create the Role to bind it to the namespace.
+		_, err = f.ClientSet.RbacV1beta1().ClusterRoles().Create(&rbacv1beta1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{Name: podSecurityPolicyPrivileged},
+			Rules: []rbacv1beta1.PolicyRule{{
+				APIGroups:     []string{"extensions"},
+				Resources:     []string{"podsecuritypolicies"},
+				ResourceNames: []string{podSecurityPolicyPrivileged},
+				Verbs:         []string{"use"},
+			}},
+		})
+		ExpectNoError(err, "Failed to create PSP role")
+	})
+
+	By(fmt.Sprintf("Binding the %s PodSecurityPolicy to the default service account in %s",
+		podSecurityPolicyPrivileged, namespace))
+	BindClusterRoleInNamespace(f.ClientSet.RbacV1beta1(),
+		podSecurityPolicyPrivileged,
+		namespace,
+		rbacv1beta1.Subject{
+			Kind:      rbacv1beta1.ServiceAccountKind,
+			Namespace: namespace,
+			Name:      "default",
+		})
+	ExpectNoError(WaitForNamedAuthorizationUpdate(f.ClientSet.AuthorizationV1beta1(),
```

Comment:
this is timing out... check for an error on the BindClusterRoleInNamespace call?

Reply:
I should have been checking for a notfound error above when creating the privileged policy, rather than checking for a non-nil return. Fixed.